src/core/providers/ogr/qgsogrproviderconnection.cpp

```diff
@@ -151,6 +231,18 @@ void QgsOgrProviderConnection::setDefaultCapabilities()
 
     if ( OGR_DS_TestCapability( hDS.get(), ODsCMeasuredGeometries ) )
       mGeometryColumnCapabilities |= GeometryColumnCapability::M;
+
+    if ( OGR_DS_TestCapability( hDS.get(), ODsCCreateLayer ) )
+      mCapabilities |= CreateVectorTable;
+
+    if ( OGR_DS_TestCapability( hDS.get(), ODsCDeleteLayer ) )
+      mCapabilities |= DropVectorTable;
+
+    if ( OGR_DS_TestCapability( hDS.get(), OLCCreateField ) )
+      mCapabilities |= AddField;
+
+    if ( OGR_DS_TestCapability( hDS.get(), OLCDeleteField ) )
+      mCapabilities |= DeleteField;
```

Comment:
those are a layer capabilities, so they should be test with OGR_L_TestCapability

Reply:
Fixed